container-app.tf

```diff
@@ -157,7 +157,7 @@ resource "azapi_resource" "default" {
   })
 
   dynamic "identity" {
-    for_each = local.container_app_identities != null ? toset([1]) : toset([])
+    for_each = local.container_app_identities != {} ? [1] : [0]
```

Comment:
`[0]` is equal to `[1]` because the array length is still `1`. This ought to be `[]` (empty) (length: 0)